src/main/resources/avro/alleleAnnotations.avdl

```diff
@@ -0,0 +1,190 @@
+ LOW,       // Low disruption of protein impact
+ MODIFIER   // No known effect
+}
+
+/**
+A VariantAnnotationSet record groups VariantAnnotation records. It is derived
+from a VariantSet and holds information describing the software and reference
+data used in the annotation.
+*/
+record VariantAnnotationSet {
+
+  /** The ID of the variant annotation set record */
+  string id;
+
+  /** The ID of the variant set to which this annotation set belongs */
+  string variantSetId;
+
+  /**
+  Analysis details. It is essential to supply versions for all software and
+  reference data used.
+  */
+  Analysis analysis;
+}
+
+/**
+A transcript effect record is a set of information describing the
+effect of an allele on a transcript
+*/
+record TranscriptEffect {
+
+  /** The ID of the transcript effect record
+  */
+  string id;
+
+  //TODO: derive unique id from digest of data [location, allele, transcript?]
+
+  /** The id of the transcript feature the annotation is relative to
+  */
+  string featureId;
+
+  /** Alternate allele - a variant may have more than one alternate allele,
+  each of which will have distinct annotation.
+  */
+  union { null, string} alternateBases = null;
+
+  /** Effect of variant on this feature */
+  array<OntologyTerm> effects;
+
+  /** Highest Impact from the predicted effects */
+  Impact impact;
+
+  /** HGVS formatted annotation at genomic level  */
+  union { null, string } HGVSg = null;
+
+  /** HGVS formatted annotation at transcript level */
+  union { null, string } HGVSc = null;
+
+  /** HGVS formatted annotation at protein level */
+  union { null, string } HGVSp = null;
+
+  /** Change relative to cDNA  */
+  union {null, AlleleLocation} cDNALocation =null;
+
+  /* Change relative to coding sequence  */
+  union { null, AlleleLocation} CDSLocation = null;
+
+  /** Change relative to protein  */
+  union { null, AlleleLocation} proteinLocation = null;
+
+  /** Output from prediction packages such as SIFT */
+  array<AnalysisResult> analysisResults;
+}
+
+/**
+A `VariantAnnotation` record represents the result of comparing a variant
+to a set of reference data.
+*/
+record VariantAnnotation {
+
+  /** The ID of this VariantAnnotation. */
+  string id;
+
+  /** The variant ID. */
+  string variantId;
+
+  /** The ID of the variant annotation set this record belongs to. */
+  string variantAnnotationSetId;
+
+  /** The date this annotation was created in milliseconds from the epoch. */
+  union { null, long } created = null;
+
+  /**
+  The transcript effect annotation for the alleles of this variant. Each one
+  represents the effect of a single allele on a single transcript.
+  */
+  array<TranscriptEffect> transcriptEffects = [];
+
+  /**
+  The IDs of other variants which are co-located with this variant.
+  these can use used to look up disease associations, ClinVar statuses,
+  allele frequencies in reference panels, etc
+  */
+  array<string> coLocatedVariants =[];
```

Comment:
I see, but it does complicate things server side quite a bit. Is it really a big enough problem that it needs to go into the first version of the schema? We can always add things to the schema later, but removing things is much, much, much harder. Performance optimisations can be added later when we have hard data proving that they are necessary...